File: src/main/java/no/novari/flyt/archive/gateway/resource/configuration/ResourcePublishingConfigurationProperties.java
package no.novari.flyt.archive.gateway.resource.configuration;

import jakarta.validation.Valid;
import lombok.Getter;
import lombok.Setter;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.validation.annotation.Validated;

@Validated
@Setter
@Getter
@ConfigurationProperties(prefix = "novari.flyt.archive.gateway.resource.publishing")
public class ResourcePublishingConfigurationProperties {

    @Valid
    private ResourcePublishingResetConfigurationProperties reset;

    @Valid
    private ResourcePublishingPullConfigurationProperties pull;
}

File: src/main/kotlin/no/novari/flyt/archive/gateway/resource/configuration/ResourcePublishingConfigurationProperties.kt
package no.novari.flyt.archive.gateway.resource.configuration

import jakarta.validation.Valid
import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.validation.annotation.Validated

@Validated
@ConfigurationProperties(prefix = "novari.flyt.archive.gateway.resource.publishing")
class ResourcePublishingConfigurationProperties {
    @field:Valid
    var reset: ResourcePublishingResetConfigurationProperties? = null

    @field:Valid
    var pull: ResourcePublishingPullConfigurationProperties? = null
}
